tests/snap_test.go

```diff
@@ -0,0 +1,244 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/canonical/matter-snap-testing/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	// enviroment variables
+	specificGpioChip = "GPIO_CHIP"
+	specificGpioLine = "GPIO_LINE"
+	gpioChipMock     = "MOCK_GPIO"
+)
+
+const snapMatterPiGPIO = "matter-pi-gpio-commander"
+const chipToolSnap = "chip-tool"
+
+const sedAuthorizeMockRead = `sed -i '/\/sys\/devices\/platform\/axi\/\*.pcie\/\*.gpio\/gpiochip4\/dev/a \      - /sys/devices/platform/gpio-mockup.*/gpiochip*/dev' squashfs-root/meta/snap.yaml`
+const sedAuthorizeMockGPIOsChips = `sed -i 's/gpiochip[0,4]/gpiochip\*/' squashfs-root/meta/snap.yaml `
+
+var start = time.Now()
+var gpioChip = os.Getenv(specificGpioChip)
+var gpioLine = os.Getenv(specificGpioLine)
+
+func TestMain(m *testing.M) {
+	teardown, err := setup()
+	if err != nil {
+		log.Fatalf("Failed to setup tests: %s", err)
+	}
+
+	code := m.Run()
+	teardown()
+
+	os.Exit(code)
+}
+
+func setup() (teardown func(), err error) {
+	var newPath string
+
+	log.Println("[CLEAN]")
+	utils.SnapRemove(nil, snapMatterPiGPIO)
+	utils.SnapRemove(nil, chipToolSnap)
+
+	log.Println("[SETUP]")
+
+	teardown = func() {
+		log.Println("[TEARDOWN]")
+		utils.SnapDumpLogs(nil, start, snapMatterPiGPIO)
+
+		utils.Exec(nil, "rm "+newPath)
+
+		log.Println("Removing installed snap:", !utils.SkipTeardownRemoval)
+		if !utils.SkipTeardownRemoval {
+			utils.SnapRemove(nil, snapMatterPiGPIO)
+			utils.SnapRemove(nil, chipToolSnap)
+			utils.Exec(nil, "./gpio-mock.sh teardown")
+		}
+	}
+
+	// setup gpio mock
+	if newPath, err = setupGPIOMock(utils.LocalServiceSnapPath); err != nil {
+		teardown()
+		return
+	}
+
+	// install matter-pi-gpio-commander
+	if utils.LocalServiceSnap() {
+		err = utils.SnapInstallFromFile(nil, newPath)
+	} else {
+		err = utils.SnapInstallFromStore(nil, snapMatterPiGPIO, utils.ServiceChannel)
+	}
+	if err != nil {
+		teardown()
+		return
+	}
+
+	if err = setupGPIO(); err != nil {
+		teardown()
+		return
+	}
+
+	// connect interfaces:
+	utils.SnapConnect(nil, snapMatterPiGPIO+":avahi-control", "")
+	utils.SnapConnect(nil, snapMatterPiGPIO+":bluez", "")
+	utils.SnapConnect(nil, snapMatterPiGPIO+":network", "")
+	utils.SnapConnect(nil, snapMatterPiGPIO+":network-bind", "")
+	utils.SnapConnect(nil, snapMatterPiGPIO+":custom-gpio", snapMatterPiGPIO+":custom-gpio-dev")
```

Comment:
Removing auto-connected interfaces:
```suggestion
	utils.SnapConnect(nil, snapMatterPiGPIO+":avahi-control", "")
	utils.SnapConnect(nil, snapMatterPiGPIO+":bluez", "")
```

Reply:
But when installed from local `.snap` file, what guarantees that this will remain auto-connected? Does the` snapd` queries the store, identify the equivalent published snap and allows the auto-connection?  Because when installing from a local file, it doesn't have the snap signatures, so we need to use the `--dangerous`.


Wouldn't be better to remove for the case that isn't installed from snap store and leave in the case that is installed from local file?

Reply:
```suggestion
	// install matter-pi-gpio-commander
	if utils.LocalServiceSnap() {
		err = utils.SnapInstallFromFile(nil, newSnapPath)
		// connect interfaces:
		utils.SnapConnect(nil, snapMatterPiGPIO+":avahi-control", "")
		utils.SnapConnect(nil, snapMatterPiGPIO+":bluez", "")
		utils.SnapConnect(nil, snapMatterPiGPIO+":network", "")
		utils.SnapConnect(nil, snapMatterPiGPIO+":network-bind", "")
		utils.SnapConnect(nil, snapMatterPiGPIO+":custom-gpio", snapMatterPiGPIO+":custom-gpio-dev")
	} else {
		err = utils.SnapInstallFromStore(nil, snapMatterPiGPIO, utils.ServiceChannel)
		// connect interfaces:
		utils.SnapConnect(nil, snapMatterPiGPIO+":avahi-control", "")
		utils.SnapConnect(nil, snapMatterPiGPIO+":bluez", "")
	}
```

Reply:
Right, the custom device needs to be connected for the local installation. Network and network-bind get auto connected regardless because these assertions come from SnapD and not the snap itself.